src/chain/tracker.rs

```
@@ -137,34 +202,44 @@ pub fn start_chain_watch(
                                 let mut id_remove_list = Vec::new();
 
                                 match transfer_events(
-                                    &client,
+                                    &subxt_client,
                                     &block,
-                                    &watcher.metadata,
-                                )
-                                    .await {
-                                        Ok((timestamp, events)) => {
-                                        tracing::debug!("Got a block with timestamp {timestamp:?} & events: {events:?}");
+                                ).await {
+                                    Ok((timestamp, extrinsics)) => {
+                                        tracing::debug!("Got a block with timestamp {timestamp:?} & {} extrinsics", extrinsics.len());
+
+                                        // TODO: handle Err results? Log them at least?
```

I think we can do one iteration here instead directly on chain.
also agree on not silently ignoring the errors